csv2vcf.py

```diff
@@ -10,28 +10,51 @@
 import csv
 import json
 
+default_input_file_format = {
+    'first_name': None,
+    'last_name': None,
+    'full_name': None,
+    'nickname': None,
+    'org': None,
+    'tel': None,
+    'url': None,
+    'bday': None,
+    'role': None,
+    'email': None,
+    'note': None
+}
+
 
 def convert_to_vcard(input_file, single_output, input_file_format):
 
-    FN = input_file_format['name']-1 if 'name' in input_file_format else None
-    NICKNAME = input_file_format['nickname']-1 if 'nickname' in input_file_format else None
-    ORG = input_file_format['org']-1 if 'org' in input_file_format else None
-    TEL = input_file_format['tel']-1 if 'tel' in input_file_format else None
-    URL = input_file_format['url']-1 if 'url' in input_file_format else None
-    BDAY = input_file_format['bday']-1 if 'bday' in input_file_format else None
-    ROLE = input_file_format['role']-1 if 'role' in input_file_format else None
-    EMAIL = input_file_format['email']-1 if 'email' in input_file_format else None
-    NOTE = input_file_format['note']-1 if 'note' in input_file_format else None
+    input_file_format = {**default_input_file_format, **input_file_format}
+
+    FIRST_NAME = input_file_format['first_name']
+    LAST_NAME = input_file_format['last_name']
+    FULL_NAME = input_file_format['full_name']
+
+    NICKNAME = input_file_format['nickname']
+
+    ORG = input_file_format['org']
+    TEL = input_file_format['tel']
+    URL = input_file_format['url']
+    BDAY = input_file_format['bday']
+    ROLE = input_file_format['role']
+    EMAIL = input_file_format['email']
+    NOTE = input_file_format['note']
 
     # if single output option is selected
-    if single_output :
-        with open( input_file, 'r' ) as source_file:
-            reader = csv.reader( source_file )
+    if single_output:
+        with open(input_file, 'r', encoding='utf-8-sig') as source_file:
+            reader = csv.DictReader(source_file)
             single_vcf = open('csv2vcf/all_contacts.vcf', 'w')
             i = 0
             for row in reader:
 
-                FN_VAL = row[FN] if FN is not None else ''
```

Comment:
new logic for building the FN_VAL based on the `full_name` or `first_name` + `last_name` pair